templates/cluster-template-external-cloud-provider.yaml

```diff
@@ -26,8 +26,8 @@ spec:
   location: ${AZURE_LOCATION}
   networkSpec:
     vnet:
-      name: ${AZURE_VNET_NAME}
-  resourceGroup: ${AZURE_RESOURCE_GROUP}
+      name: ${AZURE_VNET_NAME:=$CLUSTER_NAME-vnet}
+  resourceGroup: ${AZURE_RESOURCE_GROUP:=$CLUSTER_NAME-resource-group}
```

Comment:
I assumed this is useful, but looking for feedback

Reply:
I think in other docs/examples we've mostly used just `$CLUSTER_NAME` as the default rg name (eg. https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/789/files#diff-1bf07101845de6da98f1c948b82c5ce0L290)

I like the idea of just requiring a `CLUSTER_NAME` for the quickstart but we need to be careful that the AZURE_JSON config matches the defaults in templates, ie. https://github.com/kubernetes-sigs/cluster-api/pull/3303/files#diff-dccb64095f5c5a09d32f119446301d12R386

Reply:
maybe the default value should be just the cluster name so its matches the AZURE_JSON_B64 
${AZURE_RESOURCE_GROUP:=$CLUSTER_NAME}
${AZURE_VNET_NAME:=$CLUSTER_NAME}

at the least the resource group we can do that for

Reply:
changed the resource group to be just the cluster name `resourceGroup: ${AZURE_RESOURCE_GROUP:=$CLUSTER_NAME}`